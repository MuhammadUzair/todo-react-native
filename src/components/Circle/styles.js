import { StyleSheet, Dimensions } from 'react-native';
import { os } from '../../utils';
import { scale, verticalScale, moderateScale } from 'react-native-size-matters';
import { baseRed } from '../../assets/styles';
export default StyleSheet.create({
  circle: {
    borderWidth: 1,
    borderColor: 'rgba(0,0,0,0.2)',
    alignItems: 'center',
    justifyContent: 'center'
  },
  bigWidth: {
    width: 50,
    height: 50,
    borderRadius: 25
  },
  smallWidth: {
    width: 18,
    height: 18,
    borderRadius: 9
  }
});
