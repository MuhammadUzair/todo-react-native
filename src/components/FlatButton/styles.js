import { StyleSheet, Dimensions } from 'react-native';
import { os } from '../../utils';
const { height, width } = Dimensions.get('window');
import { scale, verticalScale, moderateScale } from 'react-native-size-matters';

export default StyleSheet.create({
  linearGradient: {
    paddingVertical: verticalScale(15),
    paddingHorizontal: scale(10)
  },
  text: {
    textAlign: 'center'
  }
});
