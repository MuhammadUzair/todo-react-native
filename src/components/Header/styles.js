import { StyleSheet, Dimensions } from 'react-native';
import { os } from '../../utils';
const { height, width } = Dimensions.get('window');

export default StyleSheet.create({
  activityIndicatorWrap: {
    backgroundColor: 'rgb(230,230,230)',
    borderRadius: 50,
    flex: 0,
    padding: 5,
    paddingLeft: os() ? 8 : 5,
    paddingTop: os() ? 8 : 5
  }
});
