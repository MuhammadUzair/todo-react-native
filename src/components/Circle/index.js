import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import {
  blueCircle,
  greenCircle,
  redCircle,
  purpleCircle,
  yellowCircle
} from '../../assets/styles';
import styles from './styles';

const getColorCode = color => {
  switch (color) {
    case 'blueCircle':
      return blueCircle;
    case 'greenCircle':
      return greenCircle;
    case 'redCircle':
      return redCircle;
    case 'purpleCircle':
      return purpleCircle;
    case 'yellowCircle':
      return yellowCircle;
    default:
      return 'white';
  }
};

export default props => {
  const { smallCircle, color, circleStyle, onPress, isActive } = props;
  return (
    <TouchableOpacity onPress={onPress && onPress}>
      <View
        style={[
          styles.circle,
          smallCircle ? styles.smallWidth : styles.bigWidth,
          { backgroundColor: getColorCode(color) },
          circleStyle && circleStyle
        ]}
        opacity={isActive ? 1 : 0.2}
      />
    </TouchableOpacity>
  );
};
