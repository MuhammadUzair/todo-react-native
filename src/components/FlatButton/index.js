import React, { Component } from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { globalStyles, baseRed } from '../../assets/styles';
import styles from './styles';
import { Button } from 'react-native-paper';
import LinearGradient from 'react-native-linear-gradient';

export default props => {
  const { text, buttonStyle, onpress, backgroundColor } = props;

  if (backgroundColor)
    return (
      <TouchableOpacity onPress={onpress && onpress}>
        <LinearGradient
          colors={['#73E186', '#4CDA64', '#73E186']}
          style={[styles.linearGradient, buttonStyle && buttonStyle]}
        >
          <Text style={[globalStyles.boldFontFace, styles.text]}>
            {' '}
            {text && text}
          </Text>
        </LinearGradient>
      </TouchableOpacity>
    );
  return (
    <Button
      mode="outlined"
      onPress={onpress && onpress}
      color={baseRed}
      style={[styles.buttonContentStyle, buttonStyle && buttonStyle]}
    >
      {text && text}
    </Button>
  );
};
