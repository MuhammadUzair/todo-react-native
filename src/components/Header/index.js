import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import { globalStyles } from '../../assets/styles';
import styles from './styles';
import LinearGradient from 'react-native-linear-gradient';

export default props => {
  const { text, buttonStyle, onpress } = props;

  return (
    <LinearGradient
      colors={['#73E186', '#4CDA64', '#73E186']}
      style={[styles.linearGradient, buttonStyle && buttonStyle]}
    >
      <Text style={[globalStyles.boldFontFace, styles.text]}>
        {' '}
        {text && text}
      </Text>
    </LinearGradient>
  );
};
