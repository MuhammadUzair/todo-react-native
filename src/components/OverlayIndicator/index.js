import React from 'react';
import { ActivityIndicator, View, StyleSheet } from 'react-native';
import { globalStyles, baseGreen } from '../../assets/styles';
import styles from './styles';

export default () => {
  return (
    <View
      style={[
        StyleSheet.absoluteFill,
        globalStyles.flexHCenter,
        globalStyles.flexVCenter
      ]}
    >
      <View
        style={[
          globalStyles.flexHCenter,
          globalStyles.flexVCenter,
          styles.activityIndicatorWrap
        ]}
      >
        <ActivityIndicator size="large" color={baseGreen} />
      </View>
    </View>
  );
};
