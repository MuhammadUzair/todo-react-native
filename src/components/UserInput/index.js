import React from 'react';
import { TextInput } from 'react-native';
import { globalStyles, baseBlue } from '../../assets/styles';
import styles from './styles';
import { Button } from 'react-native-paper';

export default props => {
  const { value, onChangeText, placeholder, inputStyle, multiline } = props;
  return (
    <TextInput
      style={[styles.userInput, inputStyle && inputStyle]}
      onChangeText={text => onChangeText && onChangeText(text)}
      value={value && value}
      placeholder={placeholder && placeholder}
      autoCorrect={false}
      underlineColorAndroid="transparent"
      multiline={multiline ? multiline : false}
    />
  );
};
