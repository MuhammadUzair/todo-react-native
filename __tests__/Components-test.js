import React from 'react';
import Header from '../src/components/Header';
import Circle from '../src/components/Circle';
import FlatButton from '../src/components/FlatButton';
import UserInput from '../src/components/UserInput';
import OverlayIndicator from '../src/components/OverlayIndicator';

import renderer from 'react-test-renderer';

test('renders correctly Header', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly Circle', () => {
  const tree = renderer.create(<Circle />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly FlatButton', () => {
  const tree = renderer.create(<FlatButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly UserInput', () => {
  const tree = renderer.create(<UserInput />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders correctly OverlayIndicator', () => {
  const tree = renderer.create(<OverlayIndicator />).toJSON();
  expect(tree).toMatchSnapshot();
});
