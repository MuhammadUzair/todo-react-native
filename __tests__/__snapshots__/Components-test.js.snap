// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Circle 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    opacity={0.2}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderColor": "rgba(0,0,0,0.2)",
          "borderWidth": 1,
          "justifyContent": "center",
        },
        Object {
          "borderRadius": 25,
          "height": 50,
          "width": 50,
        },
        Object {
          "backgroundColor": "white",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`renders correctly FlatButton 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#d63031",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 1,
      "elevation": 0,
      "minWidth": 64,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0,
    }
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "Helvetica Neue",
              "textAlign": "left",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "color": "#d63031",
                "fontFamily": "HelveticaNeue-Medium",
              },
              Object {
                "fontFamily": "HelveticaNeue-Medium",
              },
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly Header 1`] = `
<BVLinearGradient
  colors={
    Array [
      4285784454,
      4283226724,
      4285784454,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "paddingBottom": 19.61764705882353,
        "paddingHorizontal": 21.428571428571427,
        "paddingTop": 78.47058823529412,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontStyle": "normal",
          "fontWeight": "bold",
        },
        Object {
          "color": "white",
          "fontSize": 64.28571428571428,
        },
      ]
    }
  >
     
  </Text>
</BVLinearGradient>
`;

exports[`renders correctly OverlayIndicator 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
        },
        Object {
          "justifyContent": "center",
        },
        Object {
          "backgroundColor": "rgb(230,230,230)",
          "borderRadius": 50,
          "flex": 0,
          "padding": 5,
          "paddingLeft": 8,
          "paddingTop": 8,
        },
      ]
    }
  >
    <ActivityIndicator
      animating={true}
      color="#4CDA64"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;

exports[`renders correctly UserInput 1`] = `
<TextInput
  allowFontScaling={true}
  autoCorrect={false}
  multiline={false}
  onChangeText={[Function]}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "borderColor": "gray",
        "borderWidth": 1,
        "height": 40,
      },
      undefined,
    ]
  }
  underlineColorAndroid="transparent"
/>
`;
